exec/java-exec/src/main/java/org/apache/drill/exec/server/options/FallbackOptionManager.java

```diff
@@ -102,6 +125,29 @@ public void setOption(OptionValue value) {
   }
 
   @Override
+  public void deleteOption(final String name, final OptionType type) {
+    try {
+      SystemOptionManager.getValidator(name); // ensure the option exists
+    } catch (final IllegalArgumentException e) {
+      throw UserException.validationError(e)
+        .build(logger);
```

Comment:
Initially, I made that change in the ```getValidator``` function, but it seemed like UserException was one level too deep (if that makes sense). This way the calling function can do whatever it wants with the exception e.g. wrap it in a UserException for the user, or ignore it in case of a unit test.